src/animals/Crocodile.ts

```diff
@@ -1,25 +1,20 @@
-import { ReptileAnimalType } from "../customTypes";
 import { families } from "../app";
 import Reptile from "../hierarchy/Reptile";
 
-export default class Crocodile
-  extends Reptile
-{
+export default class Crocodile extends Reptile {
   private length: number;
-  private type: ReptileAnimalType = "Crocodile";
   constructor(name: string, age: number, isMale: boolean, length: number) {
     super(name, age, isMale);
     if (length <= 0) {
       throw new Error("Length must be greater than zero");
     }
     this.length = length;
   }
-  
-  
+
   giveBirth(): void {
     if (this.isMale) {
       console.log(
-        `Only Females can give birth and ${this.name} is a proud male ${this.type}`
+        `Only Females can give birth and ${this.name} is a proud male Crocodile`
```

Comment:
you can use this.class  ( or was it this.className)  to fetch the name of the class, instead of hardcoding "crocodile"